### ISSUE TITLE
Support Disabling IPv6 DNS Query

### ISSUE THREAD

**Issue by jsakwa** [aws/aws-iot-device-sdk-cpp-v2#112]
It would be nice if there were a way to disable IPv6 DNS resolution for platforms that don't support the feature. The problem occurs on some Linux platforms because: aws-common-runtime/aws-crt-cpp/aws-common-runtime/aws-c-io/source/posix/host_resolver.c will resolve names to IPv6 addresses. This can occur even if the target DNS server is IPv4. To correct this the scope of the query may be limited to IPv4 by supplying the correct flags to the struct passed into getaddrinfo

NOTE: there is a workaround for this issue on some systems:
add the following line to /etc/gai.conf
precedence ::ffff:0:0/96  100

**Comment by jsakwa** [comment]
add the following line to /etc/gai.conf
precedence ::ffff:0:0/96 100

is an adequate workaround, so this doesn't feel like something that wants to be fixed

**Comment by JonathanHenson** [comment]
Overriding OS defaults and always requesting AAAA records was intentional. The channel bootstraping runs happy-eyeballs on the results, and we have a robust load balancing scheme to make sure we use as many addresses as possible (and blacklist poorly performing endpoints as well). Some configurations, such as a vpn, don’t accurately follow the network routing rules and we wanted to handle it manually.

If this causes a problem, we’ve got a few possibilities.
1.) you can always provide your own resolve function (it’s passed in on the config struct to the resolver)
2.) we could add some flags to the Config struct that you could set to specify (ipv4 only please)

I’m hesitant to do anything that makes widespread ipv6 harder for everyone. Having a library that allows people to turn it off is one of those things. 

Thoughts?

**Comment by jsakwa** [comment]
Thanks for the quick reply -- I actually decided to close this before you responded, because it seemed like the /etc/gai.conf workaround would be adequate. However, some other folks are having some kind of similar issues: https://github.com/aws/aws-iot-device-sdk-cpp-v2/issues/109. 

TLDR: Being able to set the flags, seems like a reasonable solution. You could also check if /proc/sys/net/ipv6 exists etc. but probably a bit more involved, so I'm in favor of the flags idea, since it sounds easier to implement, and gives us the flexibility to handle it appropriately on a case by case basis.

**Comment by jmklix** [comment]
This is not something that we feel like would be a useful feature as there seams to be a workaround. If anyone in the future has a similar or related issue please open a new issue.
